diff --git a/builds/windows/ftsystem.c b/builds/windows/ftsystem.c
index 1ebadd49f..7c49158a6 100644
--- a/builds/windows/ftsystem.c
+++ b/builds/windows/ftsystem.c
@@ -266,8 +266,30 @@
       return FT_THROW( Invalid_Stream_Handle );
 
     /* open the file */
+#if !defined( WINAPI_FAMILY ) || !( WINAPI_FAMILY == WINAPI_FAMILY_PC_APP || \
+                                    WINAPI_FAMILY == WINAPI_FAMILY_PHONE_APP )
     file = CreateFileA( (LPCSTR)filepathname, GENERIC_READ, FILE_SHARE_READ,
                         NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0 );
+#else
+    WCHAR wlpFileName[MAX_PATH];
+    if ( MultiByteToWideChar( CP_UTF8, 0, filepathname, -1, wlpFileName,
+                              MAX_PATH ) == 0 )
+    {
+      file = INVALID_HANDLE_VALUE;
+    }
+    else
+    {
+      CREATEFILE2_EXTENDED_PARAMETERS CreateExParams = { 0 };
+      CreateExParams.dwSize = sizeof( CREATEFILE2_EXTENDED_PARAMETERS );
+      CreateExParams.dwFileAttributes     = FILE_ATTRIBUTE_NORMAL & 0xFFFF;
+      CreateExParams.dwFileFlags          = FILE_ATTRIBUTE_NORMAL & 0xFFF00000;
+      CreateExParams.dwSecurityQosFlags   = FILE_ATTRIBUTE_NORMAL & 0x000F0000;
+      CreateExParams.lpSecurityAttributes = NULL;
+      CreateExParams.hTemplateFile        = 0;
+      file = CreateFile2( wlpFileName, GENERIC_READ, FILE_SHARE_READ,
+                          OPEN_EXISTING, &CreateExParams );
+    }
+#endif
     if ( file == INVALID_HANDLE_VALUE )
     {
       FT_ERROR(( "FT_Stream_Open:" ));
diff --git a/src/base/ftver.rc b/src/base/ftver.rc
index a3d05b378..d70304202 100644
--- a/src/base/ftver.rc
+++ b/src/base/ftver.rc
@@ -29,11 +29,19 @@ FILEFLAGSMASK        VS_FFI_FILEFLAGSMASK
 FILEFLAGS            VS_FF_DEBUG
 #endif
 #ifdef DLL_EXPORT
-FILETYPE             VFT_DLL
+            FILETYPE VFT_DLL
+#ifdef _DEBUG
+#define FT_FILENAME  "freetyped.dll"
+#else
 #define FT_FILENAME  "freetype.dll"
+#endif
 #else
 FILETYPE             VFT_STATIC_LIB
-#define FT_FILENAME  "freetype.lib"
+#ifdef _DEBUG
+#define FT_FILENAME  "libfreetyped.dll"
+#else
+#define FT_FILENAME  "libfreetype.lib"
+#endif
 #endif
 BEGIN
   BLOCK "StringFileInfo"
